### ISSUE TITLE
[MAC] Resolution not supported

### ISSUE THREAD

**Issue by TadSV** [ValveSoftware/Dota-2#562]
I can't use 1920x1200 or any other higher resolution in fullscreen mode, everytime that i use any of the previous i just can see the left bottom quarter in the screen and i have to reset the settings when i restart the game (also, i have to use cmd+shift+esc to quit because i can't use the button in the menu).

Apart of that issue, i have fps issues with any settings in the game, unplayable.

My specs are these:
MBP 15" Retina 2015
OS X 10.11.0 "El Capitan"
Core i7 4870HQ 2.5Ghz
16gb RAM
AMD R9 M370X 2gb GDDR5
SSD 500gb

**Comment by danginsburg** [comment]
Deskop Friendly Fullscreen only goes up to the resolution of your desktop.  What resolution is that set to?  What resolutions do you see for Exclusive Fullscreen?  A screenshot might be helpful here to narrow this down.  

Regarding performance: your system is very high spec if there is poor performance on your machine we need to investigate it.  If you could post your video settings and FPS in a botmatch that would be helpful.

**Comment by TadSV** [comment]
The desktop friendly fullscreen is 1920x1200 (Like my scaled resolution in OS X).
For fullscreen i can only see from 2880x1800 and less, i usually want to use 1920x1200 for the scaled resolution or even tried 2880x1800 but i just see the left quarter fullscreen, after that i have to use force quit otherwise is impossible to close the game (then, i use -autoconfig in steam to reset everything).

Usually i get 40-65 fps average (with spikes) to 10-25 approx. in team fights.

Btw. i can't upload the screenshots because almost every file is 10-11mb in size, i made a zip but is the same problem, could i send the file to an email?

**Comment by TadSV** [comment]
Solved.

I uploaded the zip file with the screenshots to file dropper.
Here is the link: http://www.filedropper.com/archive_85

If you need another screenshot or information let me know, i keep an eye constantly in this topic.

P.S: I forgot to tell, everytime that i use an ability or someone else, my fps decrease a lot (like 20-30fps) and i get a little freeze (almost 1-2 seconds, but in team fights is really annoying and almost impossible to play).

**Comment by danginsburg** [comment]
Thanks, I'm trying to repro the resolution issue but am not having any luck.  If I setup my MBP with this desktop resolution:

<img width="670" alt="screen shot 2015-10-13 at 11 29 55 am" src="https://cloud.githubusercontent.com/assets/375707/10459561/187f819a-719e-11e5-8d2f-d6f820c14280.png">

I see this as the modes available for exclusive fullscreen:
![2015-10-13_00001](https://cloud.githubusercontent.com/assets/375707/10459566/222ee352-719e-11e5-9815-fd0a759a93b2.jpg)

And I tried switching to each one of these and they all rendered correctly.  I am not sure how to reproduce the issue you are seeing.

**Comment by TadSV** [comment]
<img width="1920" alt="screen shot 2015-10-13 at 10 42 20 am" src="https://cloud.githubusercontent.com/assets/15100190/10459854/2472fa92-7197-11e5-8755-ee77460410e7.png">
I use most 1920x1200 for the performance, but is the same problem of the left bottom quarter with both.

Are you using OS X 10.10.x "Yosemite"? Is just a doubt, i don't know if maybe is that (I bought the MBP on the same day as the release date of "El Capitan" and made a clean install immediately).

With other games as Team Fortress 2 and Counter Strike GO i dont have any issues, just with Dota 2.

P.S: Have you already updated the settings, played and restarted the game? When i change the resolution immediately sometimes i dont have the issue, but when i restart the game again i just can see the left bottom quarter.

**Comment by danginsburg** [comment]
I am on OS X 10.11 although an older model MBPr.  I have tried setting Exclusive Fullscreen resolution at various settings, starting and restarting, and don't see the 1/4 screen bug.  To be clear:  you set your desktop to 1920x1200, exclusive fullscreen to 1920x1200 in game, exit game, come back in and it shows 1/4 of the screen.  Is that right?

**Comment by TadSV** [comment]
My desktop is the whole time in 1920x1200 scaled resolution (i don't use the default settings), i use the 1920x1200 exclusive fullscreen setting in game, sometimes when i set it up (after i press apply) everything goes well, except for the performance in game with fps spikes in teamfights and lower performance in general (40-50 fps). When i quit the game and open it again i have the 1/4 screen bug, if that isn't the case, sometimes after i press apply i get the 1/4 screen immediately.

Like i have the previous bug, im forced to play with 1680x1050 in exclusive fullscreen, but the fps performance issue is still present (even with deactivating spectaculars, everything in low, teamfights fps spikes, etc.).

When i was searching for support in google, i found a topic (i dont remember if it was this site or another forum) from another user with the same MBP 15" Retina 2015 (Dedicated graphics) issue. Maybe is specific from my model considering the gpu (AMD in contrast to the previous Nvidia 15" Retina models).

With my last MBP 15" 2011 (also AMD) in the older Source 1 client (first days after it was released) i didn't have a lot of issues like in the new reborn; in contrast to the windows client, the OS X client at least in my current mac is unplayable.

**Comment by danginsburg** [comment]
Regarding perf: are you sure it is running on the AMD R9 M370X and not the Intel Iris Pro?  On a three year old MBPr with NV GTX 650M I see perf well into the 70-80fps range with your exact settings.  You can try using gfxcardstatus to force it to use the AMD GPU: https://gfx.io/ .  There is also a Graphics Setting in El Capitan for Higher Perf vs Battery Life.

**Comment by TadSV** [comment]
It is using the M370X. Is the first thing that i check after i have an issue with a game in OS X.

I have gpu switching enabled in the energy settings and usually when i play, i just have Dota 2 open (excluding steam).

P.S: I forgot, when i play or im doing something intensive with the gpu/cpu, my mac is plugged.

**Comment by danginsburg** [comment]
@TadSV Is the 1/4 screen at 1920x1200 still happening with the most recent update?  I have tried everything I can think of to repro this issue and haven't had any luck.

**Comment by TadSV** [comment]
@danginsburg Yes, i found a temporal fix for this issue, after i start the game and i get into the menu with the 1/4 screen bug, if i press cmd+H to minimize the game in the dock and open it again, the resolution goes back to normal without problem.

I have updated my OS X to 10.11.1 and i still have the same bug (even with the new updates of Dota 2), maybe is an specific issue to my AMD M370X, because is the only game that gives me the resolution problem.

**Comment by TadSV** [comment]
Reinstalled OS X El Capitan 10.11.1 and i still have the same bug. This problem seems specific to my model.
